metaflow/plugins/kfp/kfp.py

```diff
@@ -227,21 +227,71 @@ def create_kfp_pipeline_yaml(self, pipeline_file_path) -> str:
             pipeline_conf=pipeline_conf,
         )
```

Comment:
You could change the above code to use

```python
workflow: Dict = kfp.compiler.Compiler().create_workflow(pipeline_func)
```

and modify the Argo workflow dictionary in memory without having to write a file